packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -48,17 +45,26 @@ function UnconnectedNavigatorScreen(
 	props: Props,
 	forwardedRef: ForwardedRef< any >
 ) {
+	const screenId = useRef( uuid() );
 	const { children, className, path, ...otherProps } = useContextSystem(
 		props,
 		'NavigatorScreen'
 	);
 
 	const prefersReducedMotion = useReducedMotion();
-	const { location } = useContext( NavigatorContext );
-	const isMatch = location.path === escapeAttribute( path );
+	const { location, match, addScreen, removeScreen } =
+		useContext( NavigatorContext );
+	const isMatch = match === screenId.current;
 	const wrapperRef = useRef< HTMLDivElement >( null );
 
-	const previousLocation = usePrevious( location );
```

Comment:
It's not great to get the previous location using this hack because it assumes that a change in "location" is the only thing responsible for triggering the re-rendering of the component and it's clearly not the case.

Reply:
Good point, thank you for pointing it out!